### PR TITLE
[GLUTEN-2492][CH] fix posexplode cannot handle null value as argument 

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#2492)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala [hunk 1]
```diff
@@ -93,15 +93,15 @@ case class PosExplodeTransformer(
 
     val funcMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
 
-    // map_from_arrays(sequence(1, size(array_or_map)), array_or_map)
-    val mapFromArraysFuncId = ExpressionBuilder.newScalarFunction(
+    // map_from_arrays_unaligned(sequence(1, size(array_or_map)), array_or_map)
```
Comment (exmy) [inline gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:96]: 可以顺便修下这个问题 #2454, sequence(0, ...) 从 0 开始

Comment (taiyang-li) [reply gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:96]: fixed

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Comment by yangzhg** [comment]
`map_from_arrays_unaligned` is not exist in velox ?

**Comment by taiyang-li** [comment]
> `map_from_arrays_unaligned` is not exist in velox ?

yes, it's a specific function created to fix  https://github.com/oap-project/gluten/issues/2492

**Pull request merged**